### PR TITLE
Port LazyBlock tests to ListView

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

File: packages/flutter/test/widgets/list_view_fling_test.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    await tester.fling(find.byType(ListView), const Offset(0.0, -kFlingOffset), 1000.0);
+    expect(getCurrentOffset(), kFlingOffset);
+    while (tester.binding.transientCallbackCount > 0) {
+      double lastOffset = getCurrentOffset();
+      await tester.pump(const Duration(milliseconds: 20));
+      expect(getCurrentOffset(), greaterThan(lastOffset));
+    }
+  }, skip: true); // see https://github.com/flutter/flutter/issues/5339
```
Comment (Hixie) [inline packages/flutter/test/widgets/list_view_fling_test.dart]: does this still happen with ListView?

File: packages/flutter/test/widgets/scrollable_fling_test.dart [hunk 2]
```diff
@@ -67,19 +55,19 @@ void main() {
...
     expect(log, equals(<String>['tap 18']));
-    await tester.tap(find.byType(Scrollable));
+    await tester.tap(find.byType(Scrollable2));
     await tester.pump(const Duration(milliseconds: 50));
     expect(log, equals(<String>['tap 18']));
-    await tester.tap(find.byType(Scrollable));
+    await tester.tap(find.byType(Scrollable2));
     await tester.pump(const Duration(milliseconds: 50));
     expect(log, equals(<String>['tap 18', 'tap 31']));
   }, skip: Platform.isMacOS); // Skip due to https://github.com/flutter/flutter/issues/6961
```
Comment (Hixie) [inline packages/flutter/test/widgets/scrollable_fling_test.dart]: that issue is closed now, can we unskip the test?

File: packages/flutter/test/widgets/scrollable_fling_test.dart [hunk 3]
```diff
@@ -90,18 +78,18 @@ void main() {
...
+    await tester.fling(find.byType(Scrollable2), const Offset(0.0, -200.0), 1000.0);
     await tester.pump(const Duration(milliseconds: 50));
     expect(log, equals(<String>['tap 18']));
     await tester.pump(const Duration(seconds: 50));
     expect(log, equals(<String>['tap 18']));
-    await tester.tap(find.byType(Scrollable));
+    await tester.tap(find.byType(Scrollable2));
     await tester.pump(const Duration(milliseconds: 50));
     expect(log, equals(<String>['tap 18', 'tap 43']));
   }, skip: Platform.isMacOS); // Skip due to https://github.com/flutter/flutter/issues/6961
```
Comment (Hixie) [inline packages/flutter/test/widgets/scrollable_fling_test.dart]: ditto

**Comment by Hixie** [comment]
LGTM if you have checked that all the skips are still necessary

File: packages/flutter/test/widgets/list_view_fling_test.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/test/widgets/list_view_fling_test.dart]: Yes.

File: packages/flutter/test/widgets/scrollable_fling_test.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/test/widgets/scrollable_fling_test.dart]: Still busted

File: packages/flutter/test/widgets/scrollable_fling_test.dart (hunk 3 above)
Comment (abarth) [inline packages/flutter/test/widgets/scrollable_fling_test.dart]: still busted

File: packages/flutter/test/widgets/list_view_fling_test.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/test/widgets/list_view_fling_test.dart]: wild! I was really hoping it would go away, given how much code was rewritten...

Comment (abarth) [inline packages/flutter/test/widgets/list_view_fling_test.dart]: I think it's just restarting the ticker.

**Pull request merged**